### PR TITLE
support for LocalDateTime

### PR DESCRIPTION
**Pull request by jgaleotti**

### CODE REVIEW

File: e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/reservations/ReservationsEMTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    public void testRunEM() throws Throwable {
+
+        runTestHandlingFlakyAndCompilation(
+                "MongoReservationsEM",
+                "org.foo.spring.rest.mongo.MongoEMFitness",
+                10000,
+                (args) -> {
+                    args.add("--enableWeightBasedMutationRateSelectionForGene");
```
Comment (arcuri82) [inline e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/reservations/ReservationsEMTest.java:33]: why this parameter?

File: e2e-tests/spring-rest-mongo/src/main/java/com/mongo/reservations/ReservationsController.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@RestController
+@RequestMapping(path = "/reservations")
+public class ReservationsController {
+
+    @Autowired
+    private ReservationRepository reservations;
+
+    @GetMapping("findAll")
+    public ResponseEntity<Void> findAll() {
+        int status = (reservations.findAll().isEmpty()) ? 400 : 200 ;
```
Comment (arcuri82) [inline e2e-tests/spring-rest-mongo/src/main/java/com/mongo/reservations/ReservationsController.java:16]: shouldn't we explicitely check that at least one `checkInDateTime` is not null?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/reservations/ReservationsEMTest.java (hunk 1 above)
Comment (jgaleotti) [reply e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/reservations/ReservationsEMTest.java:33]: copy paste error

File: e2e-tests/spring-rest-mongo/src/main/java/com/mongo/reservations/ReservationsController.java (hunk 2 above)
Comment (jgaleotti) [reply e2e-tests/spring-rest-mongo/src/main/java/com/mongo/reservations/ReservationsController.java:16]: ok. I will add a @NotNull annotation plus some logic in the controller

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**